src/store/blockstore-level.ts

```diff
@@ -54,13 +55,17 @@ export class BlockstoreLevel implements Blockstore {
     return this.db.close();
   }
 
-  put(key: CID, val: Uint8Array, _ctx?: Options): Promise<void> {
-    return this.db.put(key.toString(), val);
+  put(key: CID, val: Uint8Array, options?: Options): Promise<void> {
+    options?.signal?.throwIfAborted();
```

Comment:
For my learning: I see this pattern throughout, I could guess, but what would be an example or scenario of why an abort signal would already be aborted upon the start of a method executing?

Reply:
perhaps something like this?
```js
let controller = new AbortController;

try {
    await doThisThing({ signal: controller.signal });
} catch {
    controller.abort();
}

doThatThing({ signal: controller.signal });
```
i.e. if the same `AbortSignal` is used by multiple things


also it's worth mentioning that it's possible to create an already aborted `AbortSignal` via `AbortSignal.abort([reason])` (and im pretty sure the other uses of `AbortSignal` (e.g. `fetch`) do it this way too, so i was trying to be consistent)

Reply:
What would be the reason for continuing with `doThatThing()` if an abort signal is already detected in the `catch` in this case? Is the `throwIfAborted()` check in every function more of a defensive programming pattern in case the caller needlessly call `doSomething({ signal: controller.signal });`?

Reply:
you're assuming that it's not possible to recover from a rejection of `doThisThing`.  `doThatThing` could be totally independent and/or the containing function could allow for that failure and still continue (and it might be easier to use the `AbortSignal` instead of having a `throw` since it centralizes where the exception can come from, instead of having more than one spot)

i think it'd be more realistic if i rewrote it like this
```js
async function doFileStuff(path, callback, { signal } = { }) {
    let file;

    try {
        file = await fileOpen(path, { signal });
        await callback(file);
     } catch { }

     if (file)
         await fileClose(file, { signal });
 }
```
we always want to call `fileClose` (assuming that `fileOpen` succeeded) unless the entire operation was aborted (e.g. if `controller.abort()` was called inside the `callback` due to some error case)

... tho thinking a bit more perhaps filesystem stuff is also not the best example, but hopefully this better demonstrates an example scenario 😅

but yeah generally speaking it's definitely more of a defensive approach 🙂

Reply:
sorry i wasnt entirely clear.  i meant that `doThatThing` was independent of `doThisThing`, not the `signal`

to use the `doFileStuff` example, what if it was called like so
```
let controller = new AbortController;
doFileStuff("foo.txt", (file) => {
    if (file.empty())
        controller.abort();
}, { signal: controller.signal })
```
then we'd explicitly want a `options?.signal?.throwIfAborted();` as the first line inside `fileClose` as otherwise we'd potentially do _some_ work, which kinda defeats the purpose of `AbortSignal`